src/chipid.c

```diff
@@ -521,9 +521,20 @@ static const struct stlink_chipid_params devices[] = {
             .bootrom_base = 0x1ff00000,
             .bootrom_size = 0x2000
         },
+        {
+            // STM32G031/041 (from RM0444)
+            .chip_id = STLINK_CHIPID_STM32_G0_CAT1,
+            .description = "G031/G041 device",
+            .flash_type = STLINK_FLASH_TYPE_G0,
+            .flash_size_reg = 0x1FFF75E0,    // Section 38.2
+            .flash_pagesize = 0x800,         // 2K (sec 3.2)
+            .sram_size = 0x2000,             // 8K (sec 2.3)
+            .bootrom_base = 0x1fff0000,
+            .bootrom_size = 0x7800           // 30K (table 3)
```

Comment:
This bootrom size seems wrong, it should be 0x2000 (8KByte)?

Reply:
Yes, with reference to RM0444 Section 2.2.2 - Table 3 this is 8 kB.